internal/services/elasticsan/elastic_san_resource_test.go

```diff
@@ -0,0 +1,247 @@
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccElasticSAN_zoneWithZRS(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_elastic_san", "test")
+	r := ElasticSANTestResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config:      r.zoneWithZRS(data),
+			ExpectError: regexp.MustCompile("zones are not supported"),
+		},
+	})
+}
+
+func TestAccElasticSAN_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_elastic_san", "test")
+	r := ElasticSANTestResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.RequiresImportErrorStep(r.requiresImport),
+	})
+}
+
+func TestAccElasticSAN_complete(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_elastic_san", "test")
+	r := ElasticSANTestResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.complete(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("total_iops").Exists(),
+				check.That(data.ResourceName).Key("total_mbps").Exists(),
+				check.That(data.ResourceName).Key("total_size_in_tib").Exists(),
+				check.That(data.ResourceName).Key("total_volume_size_in_gib").Exists(),
+				check.That(data.ResourceName).Key("volume_group_count").Exists(),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccElasticSAN_update(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_elastic_san", "test")
+	r := ElasticSANTestResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.update(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+func (r ElasticSANTestResource) Exists(ctx context.Context, clients *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	id, err := elasticsans.ParseElasticSanID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := clients.ElasticSan.Client.ElasticSans.Get(ctx, *id)
+	if err != nil {
+		return nil, fmt.Errorf("reading %s: %+v", *id, err)
+	}
+
+	return utils.Bool(resp.Model != nil), nil
+}
+
+func (r ElasticSANTestResource) basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_elastic_san" "test" {
+  name                 = "acctestes-${var.random_string}"
+  resource_group_name  = azurerm_resource_group.test.name
+  location             = azurerm_resource_group.test.location
+  base_size_in_tib     = 1
+  extended_size_in_tib = 1
```

Comment:
Since `extended_size_in_tib` is `optional`, it does not need to be defined in `basic` ?